newton/sim/builder.py

```diff
@@ -1260,6 +1284,58 @@ def add_joint_d6(
             enabled=enabled,
         )
 
+    def add_equality_constraint(
+        self,
+        constraint_type: str,
```

Comment:
Can you add constants, such as `EQ_BODY` or `EQ_JOINT` (feel free to pick a better name) as `wp.constant` similar to how we have `GEO_SPHERE` or `JOINT_PRISMATIC` Warp constants? That way the constraint type can be handled inside Warp kernels (where we don't support strings).

Reply:
Maybe put them in `sim/types.py` for now? We will move them around later, and may also convert them to be `IntEnum` eventually (which is what we are considering for all flags/constants in Newton).